src/data_management/wms_layers/WmsLayerForm.tsx

```diff
@@ -0,0 +1,357 @@
+            try{
+              JSON.parse(values.options as string)
+            } catch(e) {
+              return false;
+            }
+            return true;
+          })()}
+          errorMessage={(()=>{
+            try{
+              JSON.parse(values.options as string)
+            } catch(e) {
+              return "needs to be valid JSON";
+            }
+            return "";
+          })()}
+        />
+        <span className={formStyles.FormFieldTitle}>
+          3: Rights
+        </span>
+        <AccessModifier
+          title={'Access Modifier'}
+          name={'access_modifier'}
+          value={values.access_modifier as string}
+          valueChanged={value => handleValueChange('access_modifier', value)}
+        />
+        <CheckBox
+          title={'Shared with other organisations'}
+          name={'sharedWithCheckbox'}
+          value={values.sharedWithCheckbox as boolean}
+          valueChanged={bool => handleValueChange('sharedWithCheckbox', bool)}
+        />
+        {values.sharedWithCheckbox ? (
+          <SlushBucket
+            title={'Organisations'}
+            name={'shared_with'}
+            placeholder={'Search organisations'}
+            // str.replace is needed because we still use uuid without dashes everywhere. If possible we should remove this everywhere.
+            value={(values.shared_with as string[]).map((str)=>str.replace(/-/g, ""))}
+            choices={organisationsToSharedWith.map((organisation: any) => {
+              return {
+                display: organisation.name,
+                value: organisation.uuid
+              }
+            })}
+            valueChanged={(value: any) => handleValueChange('shared_with', value)}
+            validated
+          />
+        ) : null}
+        <SelectBox
+          title={'Organisation'}
+          name={'organisation'}
+          placeholder={'- Search and select -'}
+          value={typeof values.organisation === "string" ? (values.organisation as string).replace(/-/g, ""): (values.organisation as string) }
+          valueChanged={value => handleValueChange('organisation', value)}
+          choices={organisations.map((organisation: any) => [organisation.uuid, organisation.name])}
+          validated={values.organisation !== null && values.organisation !== ""}
+          errorMessage={"required"}
+          // this needs to be read only in some cases !?
+          readOnly={false}
+        />
+        <SelectBox
+          title={'Supplier'}
+          name={'supplier'}
+          placeholder={'- Search and select -'}
+          value={values.supplier as string}
+          valueChanged={value => handleValueChange('supplier', value)}
+          choices={supplierIds.available.map((suppl:any)=>
+            [suppl.username, suppl.username]
+          )}
+          showSearchField={true}
+          validated
+          // this needs to be read only in some cases !?
+          readOnly={false}
+        />
```

Comment:
I believe theser fields are read-only in the design

Reply:
I made it dependent on the manager role now.
Not sure if that is how it is supposed to work, but I suspect it based on:
https://nxt3.staging.lizard.net/management/users/bca7ea06e0e9403f8baeedef33f30876/
(click i-icon above "roles")